Source/DafnyCore/ProofDependencyWarnings.cs

```diff
@@ -95,6 +176,98 @@ private static void Warn(DafnyOptions dafnyOptions, ErrorReporter reporter, Proo
         }
       }
     }
+
+    if (dafnyOptions.Get(CommonOptionBag.SuggestProofRefactoring) && depManager.idsByMemberName[scopeName].Decl is Method m) {
+      // Function hiding suggestion
```

Comment:
Use extracted methods over comments that describe what a block is supposed to do.